pkg/tls/certificatemanagement/certificatebundle.go

```diff
@@ -249,12 +238,13 @@ var certFiles = []string{
 func getSystemCertificates() ([]byte, error) {
 	for _, filename := range certFiles {
 		data, err := os.ReadFile(filename)
-		if err == nil {
+		if err != nil {
+			if !os.IsNotExist(err) {
+				return nil, fmt.Errorf(fmt.Sprintf("error occurred when loading system root certificate with name %s", filename), err)
+			}
+		} else {
```

Comment:
Nit: No need for else.

Reply:
I think it is needed, if we don't have the else then we'd do the return when os.IsNotExist(err) is true.

Reply:
That's right.